Mathlib/Tactic/NthRewrite.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2022 Moritz Doll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Doll
+-/
+
+import Lean.Elab.Tactic.Rewrite
+import Lean.Elab.BuiltinTerm
+
+/-!
+# `nth_rewrite` tactic
+
+The tactic `nth_rewrite` and `nth_rw` are variants of `rewrite` and `rw` that only performs the
+`n`th possible rewrite.
+
+-/
+namespace Mathlib.Tactic
+
+open Lean Elab Tactic Meta Parser.Tactic
+
+/-- Variant of `rewriteTarget` that allows to use `Occurrences`.
+
+This def should be in Core. -/
+def rewriteTarget' (stx : Syntax) (symm : Bool) (occs : Occurrences := Occurrences.all)
+    (config : Rewrite.Config := { : Rewrite.Config }) : TacticM Unit := do
+  Term.withSynthesize <| withMainContext do
+    let e ← elabTerm stx none true
+    let r ← (← getMainGoal).rewrite (← getMainTarget) e symm occs config
+    let mvarId' ← (← getMainGoal).replaceTargetEq r.eNew r.eqProof
+    replaceMainGoal (mvarId' :: r.mvarIds)
+
+/-- Variant of `rewriteLocalDecl` that allows to use `Occurrences`.
+
+This def should be in Core. -/
+def rewriteLocalDecl' (stx : Syntax) (symm : Bool) (fvarId : FVarId)
+    (occs : Occurrences := Occurrences.all) (config : Rewrite.Config := { : Rewrite.Config }) :
+    TacticM Unit := do
+  Term.withSynthesize <| withMainContext do
+    let e ← elabTerm stx none true
+    let localDecl ← fvarId.getDecl
+    let rwResult ← (← getMainGoal).rewrite localDecl.type e symm occs config
+    let replaceResult ← (← getMainGoal).replaceLocalDecl fvarId rwResult.eNew rwResult.eqProof
+    replaceMainGoal (replaceResult.mvarId :: rwResult.mvarIds)
+
+/-- `nth_rewrite` is a variant of `rewrite` that only changes the nth occurrence of the expression
+to be rewritten.
+
+Note: The occurrences are counted beginning with `1` and not `0`, this is different than in
+mathlib3. -/
```

Comment:
Since this tactic exists mostly for the sake of porting, how about we preserve the indexing from mathlib3, leaving a note here that after the port it can be changed?

The other alternative would be to backport the indexing change, I guess.

Reply:
mathport knows how to add one to a number, it is probably in the best position to do this indexing change easily

Reply:
Okay, I've made a PR to mathport that hopefully does this.